cmd/zpool/zpool_vdev.c

```diff
@@ -1638,7 +1641,8 @@ construct_spec(nvlist_t *props, int argc, char **argv)
 		free(spares);
 	if (l2cache)
 		free(l2cache);
-	free(top);
+	if (top)
```

Comment:
This "if" is not necessary. It is safe to call free(NULL) on any modern platform. There were a couple very historical exceptions, which is where this myth comes from, but those are ancient and can be ignored.